pypdf/_xobj_image_helpers.py

```diff
@@ -195,7 +195,14 @@ def bits2byte(data: bytes, size: Tuple[int, int], bits: int) -> bytes:
         else:
             if img.mode == "1":
                 # Two values ("high" and "low").
-                assert len(lookup) == 2 * nb, len(lookup)
+                if len(lookup) != 2 * nb:
+                    if len(lookup) < 2 * nb:
+                        raise PdfReadError(f"Not enough lookup values: Expected {2 * nb}, got {len(lookup)}.")
+                    logger_warning(
+                        f"Expected {2 * nb} lookup values, got {len(lookup)}. Ignoring trailing ones.",
+                        __name__,
+                    )
```

Comment:
I would more likely  extract the trailing characters, strip all WHITESPACES and check whether the result is an empty string: if not this means the data will be corrupted and so an exception should be raised

Reply:
Thanks for the hint. I will have a look at this tomorrow - the initial change had been written before I dug deeper into why the lookup is too long.